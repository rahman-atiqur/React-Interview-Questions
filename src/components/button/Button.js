import React from "react";

const Button = ({ label }) => {
    return <button data-testid="myBtn">{label}</button>;
};

export default Button;
