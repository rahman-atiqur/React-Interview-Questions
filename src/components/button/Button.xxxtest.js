import React from "react";
import { ReactDOM } from "react-dom";
// import Sum from "./Sum";
import Button from "./Button";

import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";


// RTL
it("Testing label from button", () => {
        const { getByTestId } = render(<Button label="save" />);
        expect(getByTestId("myBtn")).toHaveTextContent("save");
        // const {getByTestId} = render(<Button label = "click me please"></Button>)
        // expect(getByTestId("myBtn")).toHaveTextContent("click me please")
    });


it("Testing label from button by getByText",() => {
        const { getByText } = render(<Button label="button1" />);
        expect(getByText("button1")).toBeInTheDocument();
    })

//Snapshot testing
it("Snapshot testing for button",()=>{
    const anyConst=renderer.create(<Button label="Hello" />).toJSON;
    expect(anyConst).toMatchSnapshot();

})

