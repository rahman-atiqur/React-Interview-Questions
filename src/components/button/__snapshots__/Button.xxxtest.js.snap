// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing for button 1`] = `[Function]`;
